### ISSUE TITLE
The scales used during model training

### ISSUE THREAD

**Issue by Serein00** [VIDA-NYU/tile2net#83]
Hi! While studying your paper and the corresponding code, I have a detail question about the multi-scale configuration to confirm with you. Section 3.2 of the paper clearly states that the model uses four scales: [0.5, 1.0, 1.5, 2.0]. However, when I checked the forward function in ocrnet.py, I found that when self.training is true (during the training phase), the code only calls the two_scale_forward method and passes in two scales: [1.0, 0.5]. I would like to ask if this discrepancy is because "the two-scale setup is used in the training phase and the four-scale setup in the inference phase", or if there is other configuration logic that I haven’t noticed yet. Looking forward to your answer!Thank you very much!

<img width="1050" height="404" alt="Image" src="https://github.com/user-attachments/assets/ce16c25a-62b8-475f-aa42-ec3cb2fd1f93" />

**Comment by Mary-h86** [comment]
Thanks for the close read.

There are two related setups in play:

1. The network **learns** a *relative, adjacent-scale* attention using **two-scale** pairs during optimization (e.g., $[1.0, 0.5]$). This is the compute-efficient way the attention operator itself is trained. (you can read more in the original paper cited)

2.  During the **training-time validation loop** (run under `eval()`/`no_grad()`), we **chain** that operator across **four scales** (e.g., `[0.5, 1.0, 1.5, 2.0]`) to compute the metrics and drive model selection/early stopping. That’s the sense in which our paper refers to using four scales “during training.” Omitting the four-scale chain in validation would change selection dynamics and final results, even though gradients are computed only on the two-scale path.

So: 
* `self.training == True` → uses the two-scale forward to optimize the relative attention.
* Validation (still part of the training process, but run under `eval()` and `no_grad`) → applies the four-scale hierarchical fusion for metrics/selection.
Put it together, the attention is **learned** via two-scale adjacency, and the **reported/selected** performance during training uses the **four-scale** evaluation chain.

**Comment by Serein00** [comment]
Thank you  for your detailed and insightful response to my previous question—it really helped clarify the distinction between the two-scale training optimization and the four-scale validation chain. I'm grateful for your time and expertise.
I have a few follow-up questions regarding image sizes and sampling in your implementation, particularly around the config.py settings and how they relate to the paper:

1. In config.py, the crop_size is set to (1024, 1024). Does this mean that after centroid sampling, the images are cropped to 1024x1024 size before being fed into HRNetV2?
2. For the random sampling part, does it also involve cropping the images?
3. In your paper, you mention using (512, 512) dimensions—what does this refer to specifically (e.g., testing, a different configuration, or something else)?
4. When training on a custom dataset, what should the image sizes in my dataset be? Do they need to be larger than (1024, 1024) to enable centroid-based cropping?

I'd greatly appreciate any clarification or pointers to relevant code sections/papers. Looking forward to your thoughts!

<img width="764" height="134" alt="Image" src="https://github.com/user-attachments/assets/db308f8b-8e00-46b0-8f1a-131d5c33d7f8" />
